test/fixtures/mock-app.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable node/no-unpublished-require */
+
+'use strict';
+
+const path = require('path');
+const assert = require('assert');
+
+const mockFs = require('mock-fs');
+const set = require('set-value');
+
+const HomeyLib = require('../..');
+
+const PNG_HEADER = Buffer.from([
+  0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a,
+  0x00, 0x00, 0x00, 0x00, 0x49, 0x48, 0x44, 0x52,
+]);
+
+// TODO: do we want to test other image types?
+// for now we assume that if png works, the other formats will too...
+function createFakePng({ width, height }) {
+  const size = Buffer.alloc(8);
+  size.writeUInt32BE(width, 0);
+  size.writeUInt32BE(height, 4);
+
+  return Buffer.concat([PNG_HEADER, size]);
+}
+
+// This manifest is the minimum which is valid at all levels
+const baseAppManifest = {
+  id: 'test.app',
+  name: { en: 'Test' },
+  description: { en: 'Test app' },
+  brandColor: '#000000',
+  version: '1.0.0',
+  sdk: 3,
+  compatibility: '>=5.0.0',
+  platforms: ['local', 'cloud'],
+  category: ['tools'],
+  images: {
+    small: '/assets/images/small.png',
+    large: '/assets/images/large.png',
+    xlarge: '/assets/images/xlarge.png',
+  },
+  author: { name: 'Athom B.V.' },
+  support: 'mailto:support@homey.app',
+};
+
+// This manifest is the minimum which is valid at all levels
+const baseDriverManifest = {
+  id: 'test',
+  name: { en: 'Test' },
+  class: 'light',
+  capabilities: ['onoff'],
+  platforms: ['local', 'cloud'],
+  connectivity: ['cloud'],
+  images: {
+    small: '/drivers/test/assets/images/small.png',
+    large: '/drivers/test/assets/images/large.png',
+    xlarge: '/drivers/test/assets/images/xlarge.png',
+  },
+};
+
+function mockApp(manifest, { files = {}, env = {} } = {}) {
```

Comment:
The idea behind this setup is that the `mockApp()` function ensures that the `baseAppManifest` and `baseDriverManifest` will validate correctly. We can then overwrite various properties to test them.

It would probably be nicer if we where able to `validateManifest(manifest)` and `validateFiles(path)` separately. Then we'd only mock the filesystem for the `validateFiles` tests. With the current setup I'd be relatively easy to change to that.